### PR TITLE
#161255197 Users should be able view/edit their profile

### PR DESCRIPTION
**Pull request by mmosoroohh**
### What does this PR do?
- Users should be able to view and edit their profile 

### Description of Task to be completed?
- Create profile route
- Create a profile component
- Create profile reducer.
- Create action creators to dispatch actions.
- Stores token and user profile in local storage on successful login

### How should this be manually tested?
- Clone this repository.
- Branch out to ft-user-able-view-edit-their-profile-161255197
- Install node and yarn
- Run ```yarn install```
- Run ```yarn start```
- Navigate to ```Profile page```

### What are the relevant pivotal tracker stories?
[#161255197](https://www.pivotaltracker.com/story/show/161255197)

### Screenshots (if appropriate)
- Profile page
screen shot 2018-12-14 at 13 46 51

- Edit Profile 
screen shot 2018-12-14 at 13 47 46

### CODE REVIEW

**Review by Tony-Ndichu — APPROVED** [review]
I like the edit profile modal, looks good to me

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+REACT_APP_API_URL=https://ah-map-staging.herokuapp.com/api/
```
Comment (Rodgers-M) [inline .env]: this file should not be pushed to git

File: README.md [hunk 2]
```diff
@@ -1,54 +0,0 @@
-# Authors Haven - A Social platform for the creative at heart.
```
Comment (Rodgers-M) [inline README.md]: what's going on here? you deleted everything in this file

File: src/components/Menu/AuthMenuItem.js [hunk 3]
```diff
@@ -29,15 +29,20 @@ const AuthMenuItem = ({ ...props }) => {
...
+                <Icon name="user" size="big" />
+              </Link>
             </div>
             <div className="item">
               <Link to="/article">
                 <Icon name="plus circle" size="big" />
               </Link>
             </div>
+            <div className="item">
+              <a href="#logout" id="logout" onClick={() => removeToken()}>
```
Comment (Rodgers-M) [inline src/components/Menu/AuthMenuItem.js:42]: use the `Link` component in place of the anchor tag

File: src/components/Profile/Ourmodal.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+import Avatar from './Avatar';
+import { AVATAR } from '../../constants';
+import { getToken } from '../../utils/auth';
+import { updateImage, saveProfile } from '../../redux/actions';
+
+class Edit extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: getToken().user.username,
```
Comment (Rodgers-M) [inline src/components/Profile/Ourmodal.js:16]: How are you getting the user from the `getToken` method?

File: src/redux/actions/profile.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+export const updateImage = payload => ({
+  type: UPDATE_IMAGE,
+  payload,
+});
+
+
+export const getProfile = username => dispatch => api.user.profile(username)
+  .then((response) => {
+    dispatch(profileFetch(response.data));
```
Comment (Rodgers-M) [inline src/redux/actions/profile.js]: this action is dispatched after a profile has been fetched, you might consider renaming it to be more descriptive. `profileFetched` maybe

File: .env (hunk 1 above)
Comment (mmosoroohh) [reply .env]: okay
ill change that

**Review by Rodgers-M — CHANGES_REQUESTED** [review]
please see comments. 

note:
when naming actions/action-creators , please name them in a way that describes the state of the action, whether is starting, in the process or complete. i.e `fetchingProfile`, `profileFetched` e.t.c

File: src/components/Profile/Ourmodal.js (hunk 4 above)
Comment (mmosoroohh) [reply src/components/Profile/Ourmodal.js:16]: yes, once a user has logged in and his/her token stored. You can identify the user using the tokens stored...

File: src/redux/actions/profile.js (hunk 5 above)
Comment (mmosoroohh) [reply src/redux/actions/profile.js]: Okay

File: README.md (hunk 2 above)
Comment (mmosoroohh) [reply README.md]: It was a mistake that I just realised apologies for that

**Review by Mirriam-Maina — COMMENTED** [review]
The changes look good to me. Maybe you can try to improve the test coverage?

**Comment by Rodgers-M** [comment]
the `.env` file is still here

**Comment by mmosoroohh** [comment]
@Rodgers-M placed  .env file in gitignore

**Review by Rodgers-M — APPROVED** [review]

**Pull request merged**